tools/link.py

```diff
@@ -769,8 +769,7 @@ def phase_linker_setup(options, state):  # noqa: C901, PLR0912, PLR0915
 
   if settings.MODULARIZE == 'instance':
     diagnostics.warning('experimental', '-sMODULARIZE=instance is still experimental. Many features may not work or will change.')
-    if options.oformat != OFormat.MJS:
-      exit_with_error('emcc: MODULARIZE instance is only compatible with .mjs output files')
+    settings.EXPORT_ES6 = 1
```

Comment:
Isn't `oformat` set to `MJS` when ever `EXPORT_ES6` is set?

In any case I think this error still needs to be here, but maybe in form of :

```
if not settings.EXPORT_ES6:
  exit_with_error('emcc: MODULARIZE instance is only comatile with ES module output')
```

Reply:
It's the opposite MJS sets EXPORT_ES6. You can still do EXPORT_ES6 with with a `.js` file.

I don't think we plan to support instance mode in non es6 mode, so maybe it's better to auto set the flag to not force the user to set both?

Reply:
I think for now I would rather force users to specify both flags.    Maybe one day we could change that when we decide to make this stuff non-experimental.

Reply:
Also, even if we did want to auto-set on setting based on another we still need to handle and catch the case where the command line is contradictory.